### PR TITLE
[editor-common] anchor - add config to handle element preview

### PR DESCRIPTION
**Pull request by noamtwix**

### CODE REVIEW

File: examples/main/shared/editor/EditorPlugins.tsx [hunk 1]
```diff
@@ -612,6 +612,15 @@ const config: RichContentEditorProps['config'] = {
     // },
     onClick: (event, url) => console.log('link clicked!', url),
     linkTypes: { anchor: true },
+    // linkTypes: {
```
Comment (shaulgo) [inline examples/main/shared/editor/EditorPlugins.tsx:615]: what about docs?

File: website/docs/plugins_api/LinkPlugin.mdx [hunk 2]
```diff
@@ -80,6 +80,27 @@ Default value:
 }
 ```
 
+### `previewElement`
+
+anchor configuration for customize the preview element that shown in anchor panel.
```
Comment (shaulgo) [inline website/docs/plugins_api/LinkPlugin.mdx:85]:
```suggestion
Anchor configuration to customize the preview elements shown in the anchor panel.
```

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: examples/main/shared/editor/EditorPlugins.tsx [hunk 3]
```diff
@@ -612,6 +612,15 @@ const config: RichContentEditorProps['config'] = {
...
+    //   anchor: {
+    //     previewElement: block => {
```
[the remark this answers is not in the document]
Comment (noamtwix) [reply examples/main/shared/editor/EditorPlugins.tsx]: send only `anchorType` and `data`

File: website/docs/plugins_api/LinkPlugin.mdx [hunk 4]
```diff
@@ -80,6 +80,31 @@ Default value:
...
+        },
+      },
+    },
+}
+```
+
+`anchorType` - block plugin type e.g. "wix-draft-plugin-image", "wix-draft-plugin-video", unstyled...
+`data` - block entity data e.g. src, config...
+`text` - block text
+`type` - block type e.g. "atomic", "unstyled
```
Comment (shaulgo) [inline website/docs/plugins_api/LinkPlugin.mdx:106]: is there need for both anchorType and type? 
Probably anchorType is enough? But in the api it's better called type?

File: website/docs/plugins_api/LinkPlugin.mdx [hunk 5]
```diff
@@ -70,7 +70,7 @@ Make sure you link only to the lines you created`)}
 
 ### `linkTypes`
```
Comment (shaulgo) [inline website/docs/plugins_api/LinkPlugin.mdx:71]:
```suggestion
### `Anchor`
```

File: packages/editor-common/web/src/Components/AnchorComponents/AnchorableElement.jsx [hunk 6]
```diff
@@ -132,8 +132,29 @@ class AnchorableElement extends PureComponent {
 
   render() {
     const { styles } = this;
+    const { dataHook, onClick, isSelected, blockPreview, block, t } = this.props;
     const { iconThumbnail, preview } = this.state;
-    const { dataHook, onClick, isSelected, t } = this.props;
+    const blockPreviewElement = blockPreview?.({
```
Comment (shaulgo) [inline packages/editor-common/web/src/Components/AnchorComponents/AnchorableElement.jsx:137]:
```suggestion
    const blockPreviewElements = blockPreview?.({
```

**Pull request merged**